src/transport/RendezvousSession.cpp

```diff
@@ -58,6 +62,13 @@ CHIP_ERROR RendezvousSession::Init(const RendezvousParameters & params, Transpor
     if (params.GetPeerAddress().GetTransportType() == Transport::Type::kBle)
 #if CONFIG_NETWORK_LAYER_BLE
     {
+#if CONFIG_DEVICE_LAYER
+        if (!mParams.IsController())
+        {
+            // Enable BLE advertisement
+            ReturnErrorOnFailure(chip::DeviceLayer::ConnectivityMgr().SetBLEAdvertisingEnabled(true));
```

Comment:
Instead of accessing global state, could we provide some delegate to the RendezvousSession to call for BLE enable/disable?

I feel a bit uncomfortable accessing globals, because Android does not have them (missing Device Layer, but it does have BLE support) and coupling reduces the ability to unit test things.

Reply:
@andy31415 let me know how the latest commit looks.